### ISSUE TITLE
[FAIL]  executor TestInfoSchemaClusterTable

### ISSUE THREAD

**Issue by tiancaiamao** [pingcap/tidb#33747]
Data race test case tracked in https://github.com/pingcap/tidb/issues/33745

**Comment by tiancaiamao** [comment]
```
==================
WARNING: DATA RACE
Read at 0x00c000b16888 by goroutine 74:
  github.com/stretchr/testify/suite.(*Suite).Require()
      /home/genius/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:42 +0x5a
  github.com/pingcap/tidb/executor_test.(*infosSchemaClusterTableSuite).setUpRPCService.func1()
      /home/genius/project/src/github.com/pingcap/tidb/executor/infoschema_cluster_table_test.go:86 +0x46

Previous write at 0x00c000b16888 by goroutine 71:
  [failed to restore the stack]

Goroutine 74 (running) created at:
  github.com/pingcap/tidb/executor_test.(*infosSchemaClusterTableSuite).setUpRPCService()
      /home/genius/project/src/github.com/pingcap/tidb/executor/infoschema_cluster_table_test.go:85 +0x5d0
  github.com/pingcap/tidb/executor_test.(*infosSchemaClusterTableSuite).SetupSuite()
      /home/genius/project/src/github.com/pingcap/tidb/executor/infoschema_cluster_table_test.go:62 +0x184
  github.com/stretchr/testify/suite.Run()
      /home/genius/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:118 +0x597
  github.com/pingcap/tidb/executor_test.TestInfoSchemaClusterTable()
      /home/genius/project/src/github.com/pingcap/tidb/executor/infoschema_cluster_table_test.go:57 +0x44
  testing.tRunner()
      /home/genius/project/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /home/genius/project/go/src/testing/testing.go:1486 +0x47

Goroutine 71 (running) created at:
  testing.(*T).Run()
      /home/genius/project/go/src/testing/testing.go:1486 +0x724
  github.com/stretchr/testify/suite.runTests()
      /home/genius/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:203 +0x198
  github.com/stretchr/testify/suite.Run()
      /home/genius/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:176 +0x969
  github.com/pingcap/tidb/executor_test.TestInfoSchemaClusterTable()
      /home/genius/project/src/github.com/pingcap/tidb/executor/infoschema_cluster_table_test.go:57 +0x44
  testing.tRunner()
      /home/genius/project/go/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /home/genius/project/go/src/testing/testing.go:1486 +0x47
==================
```

TiDB version 37d86dada8885e4fcaa1bf73c5404b86489be3fe

**Comment by tisonkun** [comment]
Resolved in #34756.
